include/bluetooth/hci_vs.h

```diff
@@ -300,6 +300,35 @@ struct bt_hci_evt_mesh_scanning_report {
 	struct bt_hci_evt_mesh_scan_report reports[0];
 } __packed;
 
+/* HCI Proprietary vendor events */
+#define BT_HCI_PROP_EVT_PREFIX                 0xFF
+
+struct bt_hci_evt_prop {
+	u8_t  prefix;
+} __packed;
+
+struct bt_hci_evt_prop_report {
+	u8_t  data_len;
+	u8_t  data[0];
+} __packed;
+
+
+#if defined(CONFIG_BT_HCI_PROP_EXT)
+/** @typedef bt_hci_prop_cb_t
+ *  @brief Callback type for proprietary vendor specific events
+ */
+typedef void bt_hci_prop_cb_t(u8_t *data, size_t data_len);
```

Comment:
I guess this would more appropriately be `const void *data` to be friendly to app-side typecasts and to make it clear the app should not try to modify the data in the buffer.